tasks/PageIteratorTask.md

```diff
@@ -6,10 +6,10 @@ For a variety of reasons, collections of entities are often split into pages and
 
 ## Requirements
 
-- Accept any collection of entities that contains a link to a next page
-- Accept a callback function that will be called for each entity in each page of the collection.  The callback function should provide a way to indicate to the iterator to stop iterating.
-- Dereference the next page link, when each of the the entities of the current page have been passed to the callback.
-
+1. Accept any collection of entities that contains a link to a next page
+2. Accept a callback function that will be called for each entity in each page of the collection.  The callback function should provide a way to indicate to the iterator to stop iterating.
+3. Dereference the next page link, when each of the the entities of the current page have been passed to the callback.
+4. Accept or provide the ability to pass information like headers, middleware options, response type in the request for the next page. For example, when using a PageIteratorTask for processing a large number of events, consumer should be able to specify the timezone preferance in the headers. 
```

Comment:
How would *response type* be used? The callback would be designed to handle the specific response type for a PageIterator of a specific collection.

Reply:
https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/1ee7bd8b1cb861fae6d9cc69ed2dcf38a749f6f7/src/GraphRequest.ts#L498. I saw that the user can specify the response type in the GraphRequest object. Basically allow information that is send in the first or the main request to be sent using PageIterators as well.

Reply:
Got it. So far, for paging scenarios, there has only been json responses. We'd use a different mechanism for getting binary pages (ranges). I don't think we need a response type.

Reply:
Okay. I removed response type from this point.